### PR TITLE
video item collection view

### LINKED ISSUE #6050: Collection view doesn't load
### What happened?

https://canary.kodadot.xyz/stmn/collection/18?collectionId=18

image

### Please reproduce in steps

Missing pictures

### Expected Behavior

Pictures loaded :)

### What browsers are you seeing the problem on?

Chrome

### At which address did you encounter bug?

_No response_

### Are you logged in?

No

### Which wallet you are using?

PolkadotJS

### Screenshots

reason? 
image

### Relevant log output

_No response_

### Payment link for reward

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #6050 
- [ ] Requires deployment <snek/rubick/worker>

https://deploy-preview-6364--koda-canary.netlify.app/stmn/collection/u-18?collectionId=u-18

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

### <samp>🤖 Generated by Copilot at 2e95ac8</samp>

This pull request adds video support for collection banners and gallery items by using new components and utilities from the `@kodadot1/brick` library and the `utils/gallery/media` module. It also refactors some existing code to improve mime type detection and code reusability.

### <samp>🤖 Generated by Copilot at 2e95ac8</samp>

> _`BackgroundMedia`_
> _Video or image, depends_
> _On `mimeType` prop_

### CODE REVIEW

File: components/collection/CollectionHeader/CollectionBanner.vue [hunk 1]
```diff
@@ -97,6 +109,16 @@ useNuxt2Meta({
...
 
+  .background-media {
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+    z-index: -1;
+  }
```
Comment (roiLeo) [inline components/collection/CollectionHeader/CollectionBanner.vue:120]: duplicated?

**Review by roiLeo — COMMENTED** [review]
Hello, small stuff:
- takes a while on first load
- are we sure we need to have video for collectionHeader avatar?
- numbers of "HEAD" query is goddam hight (`https://image-beta.w.kodadot.xyz/ipfs/QmT2yg18YvYdECQcy5VJ4NYUd1dvVUjxJEhqiKBcMgS5UF`)

**Comment by daiagi** [comment]
> Hello, small stuff:
> - takes a while on first load
> - are we sure we need to have video for collectionHeader avatar?
> - numbers of "HEAD" query is goddam hight (`https://image-beta.w.kodadot.xyz/ipfs/QmT2yg18YvYdECQcy5VJ4NYUd1dvVUjxJEhqiKBcMgS5UF`)


1.  Yes, the videos take long to load
2. @exezbcz Wdyt? 
3. Is there any choice except check item by item? 


Overall these 3 points are faces of same issue, loading many videos is expensive and takes time 

This collection would of been better represented by 1 card, since all tee nfts are duplicatates 

Cc @exezbcz

**Comment by preschian** [comment]
are we sure to load all as mp4? in this PR https://github.com/kodadot/nft-gallery/pull/6080/files#diff-20e275083e1985d241cf771341545d2beefe113ec1f323738c823fb7db94452fR22 I'm trying to load video thumbnail instead of mp4 files. crash on my side if we load all as mp4 files

**Comment by daiagi** [comment]
@preschian video thumbnails would be better. why wasn't your PR merged?

**Comment by exezbcz** [comment]
@daiagi nope, we don't have to have a video for some sort of thumbnail image.

as in #6310
> its a video, but it's not showing the thumbnail of it. Would it be possible to take frame 1 maybe and use it as a preview? If not, we can use a placeholder. The main is to maintain the square preview everywhere.

- thumbnail is good 
- I think in the future there would be a player directly in the nft card - or maybe a label of video with icon....



> This collection would of been better represented by 1 card, since all the nfts are duplicates
- yup, on the collection there is no way to represent it without some sort of listing tab in gallery item.
- on explore there is a solution -> #5385

**Comment by preschian** [comment]
> @preschian video thumbnails would be better. why wasn't your PR merged?

cold cache works fine + localStorage. but, on the warm cache,
the problem was that process to get the thumbnail video is in the client. so, the performance will depend on the client resources and how many variant videos are on that page

**Comment by daiagi** [comment]
> > @preschian video thumbnails would be better. why wasn't your PR merged?
> 
> cold cache works fine + localStorage. but, on the warm cache, the problem was that process to get the thumbnail video is in the client. so, the performance will depend on the client resources and how many variant videos are on that page

so for this collection and other like it that have a lot of duplicates, work is being done to merge the cards:

* https://github.com/kodadot/stick/pull/28


not immediately related to the loading of the video issue, but an important step to improve UX and performance

**Pull request closed without merging**